proxy/grpcproxy/watcher_single.go

```diff
@@ -53,9 +52,8 @@ func (ws watcherSingle) run() {
 	defer close(ws.donec)
 
 	for wr := range ws.ch {
-		ws.rev = wr.Header.Revision
+		ws.lastSeenRev = wr.Header.Revision
```

Comment:
but the header is just where the rev was taken from; it's not a revision of a header...